xarray/core/duck_array_ops.py

```diff
@@ -251,6 +253,9 @@ def count(data, axis=None):
 
 def where(condition, x, y):
     """Three argument where() with better dtype promotion rules."""
+    # sparse support
+    if isinstance(x, sparse_array_type) or isinstance(y, sparse_array_type):
```

Comment:
I am a little surprised this is necessary. Does `sparse` not support `__array_function__` for `np.where`?

Reply:
Well, yes. sparse looks not working with `np.result_type` and `astype(copy=False)`.
I'll add a TODO here.

Reply:
do you have the latest version of sparse installed?

when I test this on my machine, it works:
```
In [13]: import sparse

In [14]: import numpy as np

In [15]: import xarray

In [16]: x = sparse.COO(np.arange(3))

In [17]: xarray.core.duck_array_ops.where(x > 1, x, x)
Out[17]: <COO: shape=(3,), dtype=int64, nnz=2, fill_value=0>
```

Reply:
Thanks. You are right. 
I was running with sparse 0.7.0. With 0.8.0, it is running.